opencog/atomutils/AtomUtils.cc

```diff
@@ -304,4 +304,58 @@ HandleSeq get_predicates_for(const Handle& target,
     return answer;
 }
 
+bool are_similar(const Handle& h1, const Handle& h2,bool strict_type_match)
+{
+    if (h1 == h2) return true;
+
+    if (NodeCast(h1) and NodeCast(h2)) {
+        if (strict_type_match and (h1->getType() != h2->getType()) )
+            return false;
+        else
+            return true;
+    }
+
+    LinkPtr lh1(LinkCast(h1));
+    LinkPtr lh2(LinkCast(h2));
+
+    if (lh1 and lh2) {
+        if(strict_type_match and (lh1->getType() != lh2->getType()))
+        {
+         return false;
```

Comment:
Oh, I mean the indentation of 'return false;' isn't correct, and I can only assume that your editor didn't auto-correct it as you typed. What development environment do you use?